app/router/router.go

```diff
@@ -80,6 +87,80 @@ func (r *Router) PickRoute(ctx routing.Context) (routing.Route, error) {
 	return &Route{Context: ctx, outboundTag: tag}, nil
 }
 
+// AddRule implements routing.Router.
+func (r *Router) AddRule(config *serial.TypedMessage) error {
+
+	inst, err := config.GetInstance()
+	if err != nil {
+		return err
+	}
+	if c, ok := inst.(*Config); ok {
```

Comment:
I see @vrnobody made a comment that it is useful to replace all existing rules. I think it is a good point that rules relate to each other. Can we make the api accept an array of Configs here, not just one?
Then maybe we can add a flag in AddRuleRequest to control if we want to append or replace.

Reply:
> Can we make the api accept an array of Configs here, not just one?

why array of configs? we currently can pass multiple json configs to AddRule api and each config can includes number of rules as well!. ( let me know if i didn’t understood the point)

> Then maybe we can add a flag in AddRuleRequest to control if we want to append or replace.

i think we can add this flag to api. seems cool

Reply:
My bad, I see now you already pass in the whole rule list